### PR TITLE
Paper cuts

### PR DESCRIPTION
**Pull request by hendrikvanantwerpen**

### CODE REVIEW

File: src/variables.rs [hunk 1]
```diff
@@ -125,6 +125,10 @@ impl<'a> Globals<'a> {
         Self(VariableMap::new())
     }
 
+    pub fn new_child(parent: &'a mut Globals<'a>) -> Self {
```
Comment (tausbn) [inline src/variables.rs:128]: The word "child" here threw me for a bit, but I see now that this is not merely a map from variables to values, but a _tree_ of such maps. I feel like this ought to be documented more clearly, but I guess that doesn't have to be in this PR.

File: src/functions.rs [hunk 2]
```diff
@@ -128,6 +128,8 @@ impl Functions {
         functions.add(Identifier::from("plus"), stdlib::math::Plus);
         // string functions
         functions.add(Identifier::from("replace"), stdlib::string::Replace);
+        // list functions
+        functions.add(Identifier::from("is-empty"), stdlib::list::IsEmpty);
```
Comment (tausbn) [inline src/functions.rs:132]: Presumably there should also be a call that adds the `length` function?

**Review by tausbn — CHANGES_REQUESTED** [review]
A few comments, but otherwise this looks good to me. :slightly_smiling_face:

File: src/variables.rs (hunk 1 above)
Comment (hendrikvanantwerpen) [reply src/variables.rs:128]: Good point! I can definitely add a comment. If you have a suggestion for a better name, I'm also open to that.

File: src/functions.rs (hunk 2 above)
Comment (hendrikvanantwerpen) [reply src/functions.rs:132]: Yes, there should!

File: src/variables.rs (hunk 1 above)
Comment (hendrikvanantwerpen) [reply src/variables.rs:128]: For example, `new_nested` or `new_inherit` (or `new_inheriting`?).

Comment (tausbn) [reply src/variables.rs:128]: Hmm... `new_environment`, perhaps? Though I do like the explicitness of saying it's nested/inherited. One could also use the word "scope", though I feel like that's potentially overloaded already in this context.

**Review by tausbn — APPROVED** [review]

**Pull request merged**